### DISCUSSION TITLE
SDK oAuth login example with JWT

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#9954] [Archive]
How to login with an Access_Token using SDK Client without credentials ?

**Comment by rijkvanzanten** [comment]
You can't retrieve a new refresh token from an active access token. There's no way to login to the API using just the access token

**Reply by NilsBaumgartner1994** [reply]
Sorry i meant the directus_access_token not the access_token itself. But i found just needed to set the refresh_token and auth_token manually.
This would be nice to give it/these to a directus sdk instance as parameters.

**Reply by NilsBaumgartner1994** [reply]
My solution was to create a custom storage, set access_token and refresh_token, and create the directus instance with this storage.

**Reply by rijkvanzanten** [reply]
@joselcvarela thoughts on the above? Would be nice to have a method to set the access / refresh tokens manually if you fetched them from an external source 🤔  The SDK could take over the auto-refreshing etc  after that

**Reply by NilsBaumgartner1994** [reply]
Yeah thats what i implemented.


    set auth_refresh_token(token) {
        this.set(KEY_AUTH_REFRESH_TOKEN, token);
    }
    get auth_refresh_token() {
        return localStorage.getItem(KEY_AUTH_REFRESH_TOKEN);
    }

the rest was copied from the storage of the sdk storage implementation

**Reply by NilsBaumgartner1994** [reply]
https://github.com/directus/directus/discussions/9984
but i found the implementation already

https://github.com/directus/directus/blob/main/packages/sdk/src/base/storage/base.ts
